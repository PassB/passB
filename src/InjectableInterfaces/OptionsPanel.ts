import {ComponentType} from 'react';
import {MapTypeAllowedData, TypedMap} from 'State/Types/TypedMap';

export interface OptionPanelProps<OptionType extends MapTypeAllowedData<OptionType>> {
  options: TypedMap<OptionType>;
  updateOptions: (newOptions: TypedMap<OptionType>) => void;
}

export type OptionsPanelType<OptionType extends MapTypeAllowedData<OptionType>> = ComponentType<OptionPanelProps<OptionType>>;
