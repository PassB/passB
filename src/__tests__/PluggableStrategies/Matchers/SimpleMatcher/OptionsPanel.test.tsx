import {mount, shallow, ShallowWrapper} from 'enzyme';
import * as React from 'react';
import {OptionsPanel} from 'PluggableStrategies/Matchers/SimpleMatcher/OptionsPanel';
import {Options} from 'PluggableStrategies/Matchers/SimpleMatcher/SimpleMatcher';
import {createTypedMap, TypedMap} from 'State/Types/TypedMap';

const getOptions = (partial: Partial<Options> = {}) => createTypedMap({
  autoWww: true,
  matchInSubDirs: false,
  ignoreLastPart: false,
  requireFullMatch: false,
  ...partial,
});

describe('SimpleMatcher OptionsPanel', () => {
  describe('rendering', () => {
    function testRendering(testOptions: TypedMap<Options>): void {
      let panel: ShallowWrapper<{}>;
      it('renders', () => {
        panel = shallow(
          <OptionsPanel options={testOptions} updateOptions={() => 0}/>,
        );
        expect(panel).toMatchSnapshot();
      });

      it('autoWww is checked correctly', () => {
        expect(
          (panel.find('#autoWww').prop('control') as JSX.Element).props.checked,
        ).toBe(testOptions.get('autoWww'));
      });

      it('matchInSubDirs is checked correctly', () => {
        expect(
          (panel.find('#matchInSubDirs').prop('control') as JSX.Element).props.checked,
        ).toBe(testOptions.get('matchInSubDirs'));
      });

      it('ignoreLastPart is checked correctly', () => {
        expect(
          (panel.find('#ignoreLastPart').prop('control') as JSX.Element).props.checked,
        ).toBe(testOptions.get('ignoreLastPart'));
      });

      it('requireFullMatch is checked correctly', () => {
        expect(
          (panel.find('#requireFullMatch').prop('control') as JSX.Element).props.checked,
        ).toBe(testOptions.get('requireFullMatch'));
      });

    }

    describe('default values', () => testRendering(getOptions()));
    describe('autoWww false', () => testRendering(getOptions({autoWww: false})));
    describe('matchInSubDirs true', () => testRendering(getOptions({matchInSubDirs: true})));
    describe('ignoreLastPart true', () => testRendering(getOptions({ignoreLastPart: true})));
    describe('requireFullMatch true', () => testRendering(getOptions({requireFullMatch: true})));
  });

  describe('should update options on interaction', () => {
    const optionsCallback = jest.fn();

    const testOptions: TypedMap<Options> = getOptions();

    const panel = mount(
      <OptionsPanel options={testOptions} updateOptions={optionsCallback}/>,
    );

    beforeEach(() => {
      optionsCallback.mockReset();
    });

    it('updates autoWww', () => {
      panel.find('#autoWww').find('input').hostNodes()
        .simulate('change', {target: {checked: !testOptions.get('autoWww')}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('autoWww', !testOptions.get('autoWww')));
    });

    it('updates matchInSubDirs', () => {
      panel.find('#matchInSubDirs').find('input').hostNodes()
        .simulate('change', {target: {checked: !testOptions.get('matchInSubDirs')}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('matchInSubDirs', !testOptions.get('matchInSubDirs')));
    });

    it('updates ignoreLastPart', () => {
      panel.find('#ignoreLastPart').find('input').hostNodes()
        .simulate('change', {target: {checked: !testOptions.get('ignoreLastPart')}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('ignoreLastPart', !testOptions.get('ignoreLastPart')));
    });

    it('updates requireFullMatch', () => {
      panel.find('#requireFullMatch').find('input').hostNodes()
        .simulate('change', {target: {checked: !testOptions.get('requireFullMatch')}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('requireFullMatch', !testOptions.get('requireFullMatch')));
    });
  });

});
