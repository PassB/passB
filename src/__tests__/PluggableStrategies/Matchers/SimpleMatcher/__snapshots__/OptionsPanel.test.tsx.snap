// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleMatcher OptionsPanel rendering autoWww false renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "autoWww": false,
        "matchInSubDirs": false,
        "ignoreLastPart": false,
        "requireFullMatch": false,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="autoWww"
          label="Ignore wwww. prefix"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="matchInSubDirs"
          label="Also match in subdirectories"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="ignoreLastPart"
          label="Ignore the last part of the password path (if you are using it as your login name)"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="requireFullMatch"
          label="Require exact match"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "autoWww",
          "label": "Ignore wwww. prefix",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "matchInSubDirs",
          "label": "Also match in subdirectories",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "ignoreLastPart",
          "label": "Ignore the last part of the password path (if you are using it as your login name)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "requireFullMatch",
          "label": "Require exact match",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="autoWww"
            label="Ignore wwww. prefix"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="matchInSubDirs"
            label="Also match in subdirectories"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="ignoreLastPart"
            label="Ignore the last part of the password path (if you are using it as your login name)"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="requireFullMatch"
            label="Require exact match"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "autoWww",
            "label": "Ignore wwww. prefix",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "matchInSubDirs",
            "label": "Also match in subdirectories",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "ignoreLastPart",
            "label": "Ignore the last part of the password path (if you are using it as your login name)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "requireFullMatch",
            "label": "Require exact match",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SimpleMatcher OptionsPanel rendering default values renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "autoWww": true,
        "matchInSubDirs": false,
        "ignoreLastPart": false,
        "requireFullMatch": false,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="autoWww"
          label="Ignore wwww. prefix"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="matchInSubDirs"
          label="Also match in subdirectories"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="ignoreLastPart"
          label="Ignore the last part of the password path (if you are using it as your login name)"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="requireFullMatch"
          label="Require exact match"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "autoWww",
          "label": "Ignore wwww. prefix",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "matchInSubDirs",
          "label": "Also match in subdirectories",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "ignoreLastPart",
          "label": "Ignore the last part of the password path (if you are using it as your login name)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "requireFullMatch",
          "label": "Require exact match",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="autoWww"
            label="Ignore wwww. prefix"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="matchInSubDirs"
            label="Also match in subdirectories"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="ignoreLastPart"
            label="Ignore the last part of the password path (if you are using it as your login name)"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="requireFullMatch"
            label="Require exact match"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "autoWww",
            "label": "Ignore wwww. prefix",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "matchInSubDirs",
            "label": "Also match in subdirectories",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "ignoreLastPart",
            "label": "Ignore the last part of the password path (if you are using it as your login name)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "requireFullMatch",
            "label": "Require exact match",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SimpleMatcher OptionsPanel rendering ignoreLastPart true renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "autoWww": true,
        "matchInSubDirs": false,
        "ignoreLastPart": true,
        "requireFullMatch": false,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="autoWww"
          label="Ignore wwww. prefix"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="matchInSubDirs"
          label="Also match in subdirectories"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="ignoreLastPart"
          label="Ignore the last part of the password path (if you are using it as your login name)"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="requireFullMatch"
          label="Require exact match"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "autoWww",
          "label": "Ignore wwww. prefix",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "matchInSubDirs",
          "label": "Also match in subdirectories",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "ignoreLastPart",
          "label": "Ignore the last part of the password path (if you are using it as your login name)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "requireFullMatch",
          "label": "Require exact match",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="autoWww"
            label="Ignore wwww. prefix"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="matchInSubDirs"
            label="Also match in subdirectories"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="ignoreLastPart"
            label="Ignore the last part of the password path (if you are using it as your login name)"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="requireFullMatch"
            label="Require exact match"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "autoWww",
            "label": "Ignore wwww. prefix",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "matchInSubDirs",
            "label": "Also match in subdirectories",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "ignoreLastPart",
            "label": "Ignore the last part of the password path (if you are using it as your login name)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "requireFullMatch",
            "label": "Require exact match",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SimpleMatcher OptionsPanel rendering matchInSubDirs true renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "autoWww": true,
        "matchInSubDirs": true,
        "ignoreLastPart": false,
        "requireFullMatch": false,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="autoWww"
          label="Ignore wwww. prefix"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="matchInSubDirs"
          label="Also match in subdirectories"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="ignoreLastPart"
          label="Ignore the last part of the password path (if you are using it as your login name)"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="requireFullMatch"
          label="Require exact match"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "autoWww",
          "label": "Ignore wwww. prefix",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "matchInSubDirs",
          "label": "Also match in subdirectories",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "ignoreLastPart",
          "label": "Ignore the last part of the password path (if you are using it as your login name)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "requireFullMatch",
          "label": "Require exact match",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="autoWww"
            label="Ignore wwww. prefix"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="matchInSubDirs"
            label="Also match in subdirectories"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="ignoreLastPart"
            label="Ignore the last part of the password path (if you are using it as your login name)"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="requireFullMatch"
            label="Require exact match"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "autoWww",
            "label": "Ignore wwww. prefix",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "matchInSubDirs",
            "label": "Also match in subdirectories",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "ignoreLastPart",
            "label": "Ignore the last part of the password path (if you are using it as your login name)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "requireFullMatch",
            "label": "Require exact match",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SimpleMatcher OptionsPanel rendering requireFullMatch true renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "autoWww": true,
        "matchInSubDirs": false,
        "ignoreLastPart": false,
        "requireFullMatch": true,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="autoWww"
          label="Ignore wwww. prefix"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="matchInSubDirs"
          label="Also match in subdirectories"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="ignoreLastPart"
          label="Ignore the last part of the password path (if you are using it as your login name)"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="requireFullMatch"
          label="Require exact match"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "autoWww",
          "label": "Ignore wwww. prefix",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "matchInSubDirs",
          "label": "Also match in subdirectories",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "ignoreLastPart",
          "label": "Ignore the last part of the password path (if you are using it as your login name)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "requireFullMatch",
          "label": "Require exact match",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="autoWww"
            label="Ignore wwww. prefix"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="matchInSubDirs"
            label="Also match in subdirectories"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="ignoreLastPart"
            label="Ignore the last part of the password path (if you are using it as your login name)"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="requireFullMatch"
            label="Require exact match"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "autoWww",
            "label": "Ignore wwww. prefix",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "matchInSubDirs",
            "label": "Also match in subdirectories",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "ignoreLastPart",
            "label": "Ignore the last part of the password path (if you are using it as your login name)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "requireFullMatch",
            "label": "Require exact match",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
